src/app/Services/UserService.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\User;
+use App\Models\Post;
+
+class UserService
+{
+    /**
+     * Get the user and posts.
+     *
+     * @return array<string, mixed>
+     */
+    public function getUserProfile(User $user): array
+    {
+        /** @var User $user */
+        $user = auth()->user();
```

Comment:
I think you might have miswrote this since you already `$user` instance on the controller. Why did you use `auth()->user()` and set the param type for this method to `User` model instance twice?

Reply:
I updated my PR po, sir. Thank you!